maxim/logger/smolagents/client.py

```
@@ -0,0 +1,539 @@
+        finally:
+            # Reset generation context var
+            try:
+                _current_generation_id.reset(gen_ctx_token)
+            except Exception:
+                pass
+            scribe().debug(f"――― End: {base_op_name} ―――\n")
+    
+    return smolagents_wrapper
+
+
+def instrument_smolagents(maxim_logger_instance: Logger, debug: bool = False):
```

Ruff (0.12.2)

`instrument_smolagents` is too complex (15 > 10)

(C901)